test/models.ts

```diff
@@ -0,0 +1,39 @@
+import type { Random } from "@reside-ic/random";
+
+import type { base, BaseType } from "../src/base";
+import type { DustModel, DustModelInfo } from "../src/model";
+import type { InternalStorage, Pars } from "../src/pars";
+
+// In contrast with the equivalent models in odin-js I'm leaving these
+// as TypeScript (not JS) for now to help get types right. That does
+// mean that there's a bit of a fight with extracting things from the
+// InternalStorage type.
+export class Walk implements DustModel {
+    private readonly internal: InternalStorage;
+
+    constructor(base: BaseType, pars: Pars) {
+        this.internal = {};
+        base.pars.setParScalar(pars, "n", this.internal, 1);
+        base.pars.setParScalar(pars, "sd", this.internal, 1);
+    }
+
+    public size(): number {
+        return this.internal.n as number;
+    }
+
+    public info(): DustModelInfo {
+        return [{dim: [1], length: this.internal.n as number, name: "x"}];
```

Comment:
shouldn't `dim` be `this.internal.n` here?

Reply:
yes, it should :man_facepalming: